src/utils/index.ts

```diff
@@ -1,15 +1,53 @@
 import axios, { AxiosError } from "axios";
-import { GELATO_ADDRESSES } from "../constants";
+import { W3F_API_ENDPOINT, W3F_API_ENDPOINT_DEV } from "../constants";
+import { W3fNetwork } from "../types/W3fNetworks.interface";
 
-export function isAutomateSupported(chainId: number): boolean {
-  return Boolean(GELATO_ADDRESSES[chainId]);
+class W3FApi {
+  private _network: Map<number, string> = new Map();
+
+  public async getNetwork(
+    chainId: number,
+    isDevelopment?: boolean,
+  ): Promise<W3fNetwork | null> {
+    try {
+      const cacheNetwork = this._network.get(chainId);
```

Comment:
the cache key should be different between dev and non dev.
eg `80002_dev` & `80002_prod`